### ISSUE TITLE
Use viz.json for named map in the embeds

### ISSUE THREAD

**Issue by alonsogarciapablo** [CartoDB/cartodb#7093]
Right now we're using the viz.json both in the [editor](https://github.com/CartoDB/cartodb/blob/55c3a1a1f2edd12f69609390fa51dcf138f3c825/lib/assets/javascripts/cartodb3/editor.js#L66) and the [embed](https://github.com/CartoDB/cartodb/blob/55c3a1a1f2edd12f69609390fa51dcf138f3c825/lib/assets/javascripts/cartodb3/public_editor.js#L2) to load/display a map. In the editor, we [are setting the API_KEY](https://github.com/CartoDB/cartodb/blob/55c3a1a1f2edd12f69609390fa51dcf138f3c825/lib/assets/javascripts/cartodb3/editor.js#L67) so that requests to the Maps API can be authenticated. In the embed, we're not passing the API KEY (because we can't expose it the HTML/JS) and the Maps API is returning the following error when trying to instantiate an anonymous map:

```
Error creating the map instance on Windshaft: permission denied for relation cdb_analysis_catalog
```

Given that all maps have a "named map", we can use it in the embeds to solve this issue.

Right now the viz.json is generated based on the visibility of the map/datasets, so we'd need a way to force the generation of viz.json with the information about the named map.

I think this needs to be discussed before we start working on it? Thoughts?

cc: @xavijam @viddo @javierarce @juanignaciosl @rochoa

**Comment by juanignaciosl** [comment]
I think that adding the named map information at the vizjson makes sense, and we'd already spoken about the need for this for analyses.

This is an example template:

```ruby
{:version=>"0.0.1",
 :name=>"tpl_3336135e_ffd9_11e5_acf4_080027880ca6",
 :auth=>{:method=>"open"},
 :placeholders=>{:layer0=>{:type=>"number", :default=>1}},
 :layergroup=>
  {:layers=>
    [{:type=>"http",
      :options=>
       {:urlTemplate=>
         "http://{s}.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png",
        :subdomains=>"abcd"}},
     {:type=>"cartodb",
      :options=>
       {:sql=>
         "SELECT * FROM (select * from twitter_t3chfest_reduced_1) AS wrapped_query WHERE <%= layer0 %>=1",
        :layer_name=>"twitter_t3chfest_reduced_1",
        :cartocss=>
         "/** simple visualization */\n\n#twitter_t3chfest_reduced_1{\n  marker-fill-opacity: 0.9;\n  marker-line-color: #FFF;\n  marker-line-width: 1;\n  marker-line-opacity: 1;\n  marker-placement: point;\n  marker-type: ellipse;\n  marker-width: 10;\n  marker-fill: #FF6600;\n  marker-allow-overlap: true;\n}",
        :cartocss_version=>"2.1.1",
        :interactivity=>"cartodb_id",
        "table_name"=>"\"\"."}},
     {:type=>"http",
      :options=>
       {:urlTemplate=>
         "http://{s}.basemaps.cartocdn.com/light_only_labels/{z}/{x}/{y}.png",
        :subdomains=>"abcd"}}],
   :stat_tag=>"3336135e-ffd9-11e5-acf4-080027880ca6"},
 :view=>
  {:zoom=>4,
   :center=>{:lng=>-5.625, :lat=>31.840232667909365},
   :bounds=>
    {:west=>-68.994140625,
     :south=>-16.55196172197251,
     :east=>57.74414062500001,
     :north=>64.01449619484472}}}
```

I'll wait for you to agree to begin.

**Comment by alonsogarciapablo** [comment]
Instead of adding more information to the viz.json... can't we just force the embed to use the viz.json for the named map? (i.e: generate the viz.json as if the visualization was password protected or had some private datasets [here](https://github.com/CartoDB/cartodb/blob/master/app/controllers/carto/editor/public/embeds_controller.rb#L20)).

**Comment by juanignaciosl** [comment]
Ah, sure. I don't know the inners of the generation, if behaving like a named map is good enough I can do it very easily.

If "always generate named map vizjson for new embeds" is what you need I can branch and do it for you to test.

**Comment by juanignaciosl** [comment]
Closed by #7116.
